tools/schemacode/bidsschematools/tests/conftest.py

```diff
@@ -35,23 +37,21 @@ def get_gitrepo_fixture(url, whitelist):
     @pytest.fixture(scope="session")
     def fixture():
         archive_name = url.rsplit("/", 1)[-1]
-        testdata_dir = os.path.join(os.path.dirname(__file__), "data", "bundled", archive_name)
-        testdata_dir = os.path.abspath(testdata_dir)
-        if os.path.isdir(testdata_dir):
+        testdata_dir = files("bidsschematools.tests.data") / archive_name
```

Comment:
what's wrong with `os.path`? It seems it's accomplishing the same thing just adding variant imports and potentially breaking older python compatibility.

Reply:
`__file__` is discouraged as it won't work correctly in a zipped package. See https://importlib-resources.readthedocs.io/en/latest/using.html?highlight=__file__#example for discussion. `importlib.resources` will auto-expand files to a temporary directory if needed to ensure that you can work with them, or work just like `__file__ + os.path` under the hood if the package is not zipped.

Reply:
Are we realistically expecting this to be in a zipfile... ever? But might be more important to expedite this.

Reply:
The problem is that applications might get distributed as `.pyz` files. If any of their dependencies use `__file__`, they could break in annoying ways.

This is admittedly less of a problem for tests, and I'm not super hardline about it, but when I can get it working, it seems like a win to use the approved API.